File: scratch/ant/BSF/repository/conf/sample/xmethodsStockquote.rb
<x><![CDATA[

require 'rexml/document'
include REXML

def mediateIn(msg) 
   newRequest = Document.new($REQUEST)
   newRequest.root.elements[1].text = msg.getPayloadXML().root.elements[1].elements[1].get_text
   msg.setPayloadXML(newRequest)
end

def mediateOut(msg) 
   newResponse = Document.new($RESPONSE)
   newResponse.root.elements[1].elements[1].text = msg.getPayloadXML().root.elements[1].get_text
   msg.setPayloadXML(newResponse)
end


$REQUEST = <<EOF
   <ns1:getQuote xmlns:ns1="urn:xmethods-delayed-quotes" xmlns:xsi="http://www.w3.org/1999/XMLSchema-instance" xmlns:xsd="http://www.w3.org/1999/XMLSchema">
      <symbol xsi:type="xsd:string"></symbol>
   </ns1:getQuote>
EOF

$RESPONSE = <<EOF
   <ns:getQuoteResponse xmlns:ns="http://services.samples/xsd">
      <ns:return>
         <ns:last></ns:last>
      </ns:return>
   </ns:getQuoteResponse>
EOF

]]></x>

File: scratch/synapse-2.1-versioned/scratch/ant/BSF/repository/conf/sample/xmethodsStockquote.js
<x><![CDATA[

  function mediateIn(msg) {

     var payload = new XML(msg.getPayloadXML());
     var newPayload = <ns1:getQuote xmlns:ns1="urn:xmethods-delayed-quotes" xmlns:xsi="http://www.w3.org/1999/XMLSchema-instance" xmlns:xsd="http://www.w3.org/1999/XMLSchema">
                         <symbol xsi:type="xsd:string"> { payload..*::symbol.toString() } </symbol>
                      </ns1:getQuote>;
     msg.setPayloadXML(newPayload);
  }


  function mediateOut(msg) {

     var payload = new XML(msg.getPayloadXML());

     var newPayload = <ns:getQuoteResponse xmlns:ns="http://services.samples/xsd">
                         <ns:return>
                            <ns:last> { payload..Result.toString() } </ns:last>
                         </ns:return>
                      </ns:getQuoteResponse>;

     msg.setPayloadXML(newPayload);
  }

]]></x>


